src/Ilios/AuthenticationBundle/RelationshipVoter/CurriculumInventoryInstitution.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Ilios\AuthenticationBundle\RelationshipVoter;
+
+use Ilios\AuthenticationBundle\Classes\SessionUserInterface;
+use Ilios\CoreBundle\Entity\CurriculumInventoryInstitutionInterface;
+use Ilios\CoreBundle\Entity\DTO\CurriculumInventoryInstitutionDTO;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+class CurriculumInventoryInstitution extends AbstractVoter
+{
+    protected function supports($attribute, $subject)
+    {
+        return (
+            ($subject instanceof CurriculumInventoryInstitutionDTO && in_array($attribute, [self::VIEW])) or
+            ($subject instanceof CurriculumInventoryInstitutionInterface && in_array($attribute, [
+                    self::CREATE, self::VIEW, self::EDIT, self::DELETE
+                ]))
+        );
+    }
+
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+        if (!$user instanceof SessionUserInterface) {
+            return false;
+        }
+        if ($user->isRoot()) {
+            return true;
+        }
+
+        if ($subject instanceof CurriculumInventoryInstitutionDTO) {
+            return $this->voteOnDTO($user, $subject);
+        }
+
+        if ($subject instanceof CurriculumInventoryInstitutionInterface) {
```

Comment:
This feels to me like something only root should be able to do. Any reason not to simplify it that way?

Reply:
these are just another aspect of school-level config, not application-wide config. school admins could/should be able to modify these.